mozci/push.py

```diff
@@ -1009,9 +1022,21 @@ def get_regressions(
         ):
             return regressions
 
-        for name, (count, failure_summary) in self.get_candidate_regressions(
-            runnable_type
-        ).items():
+        if build:
```

Comment:
All this code block should be in `get_candidate_regressions` provided you pass along the new build parameter.

That way the `get_regressions` method remains almost unchanged